pkg/validate/docker_validator.go

```diff
@@ -0,0 +1,197 @@
+	passwordFileProvided := len(registry.PasswordFile) != 0
+
+	if passwordProvided && passwordFileProvided || passwordCommandProvided && passwordProvided || passwordCommandProvided && passwordFileProvided {
+		return false, errors.New("You have provided more than one of password, password command, or password file for your docker registry. You can specify at most one.")
+	}
+
+	if passwordProvided {
+		password, err := inspect.GetObjectPropString(ctx, registry, "password")
+		if err == nil {
+			resolvedPassword = password
+		}
+	} else if passwordFileProvided {
+		pf, err := inspect.GetObjectPropString(ctx, registry, "PasswordFile")
+		if err != nil {
+			return false, err
+		}
+		password, err := d.loadPasswordFromFile(pf, ctx, spinSvc.GetSpinnakerConfig())
+
+		if err == nil {
+			resolvedPassword = password
+		}
+		if len(resolvedPassword) == 0 || err != nil {
+			return false, errors.New("The supplied password file is empty.")
+		}
+	} else if passwordCommandProvided {
+		out, err := exec.Command("bash", "-c", registry.PasswordCommand).Output()
+
+		if err != nil {
+			return false, errors.New(fmt.Sprintf("password command returned non 0 return code, stderr/stdout was: %s", err))
+		}
+
+		resolvedPassword = strings.Trim(string(out), "\n")
+		if len(resolvedPassword) == 0 {
+			return false, fmt.Errorf("Resolved password was empty, missing dependencies for running password command?")
+		}
+
+	}
+
+	if len(resolvedPassword) != 0 && len(registry.Username) == 0 {
+		return false, errors.New("You have supplied a password but no username.")
+	} else if len(resolvedPassword) == 0 && len(registry.Username) != 0 {
+		return false, errors.New("You have a supplied a username but no password.")
+	}
+
+	service := dockerRegistryService{address: registry.GetAddress(), username: registry.Username, password: resolvedPassword, httpService: util.HttpService{}}
+
+	ok, err := service.GetBase()
+
+	if err != nil {
+		return false, err
+	}
+
+	if !ok {
+		if len(resolvedPassword) != 0 {
+			c := resolvedPassword[len(resolvedPassword)-1]
+			if unicode.IsSpace(rune(c)) {
+				return false, errors.New("Your password file has a trailing newline; many text editors append a newline to files they open." + " If you think this is causing authentication issues, you can strip the newline with the command:\n\n" + " tr -d '\\n' < PASSWORD_FILE | tee PASSWORD_FILE")
+			}
+		}
+
+		return false, errors.New(fmt.Sprintf("Unable to establish a connection with docker registry %s with provided credentials", registry.GetAddress()))
+	}
+
+	if len(registry.Repositories) != 0 {
+		for _, repository := range registry.Repositories {
+			tagCount, err := service.GetTags(repository)
+
+			if err != nil {
+				return false, err
+			}
+
+			if tagCount == 0 {
+				return false, errors.New(fmt.Sprintf("Repository %s contain any tags. Spinnaker will not be able to deploy any docker images, Push some images to your registry.", repository))
```

Comment:
Use `fmt.Errorf`. Also errors usually start with lower case in Go.

Reply:
Yeah, As this is a message to the user, I'm not sure if we need to follow that convention. What do you think?